tests/test_user_actions.py

```diff
@@ -0,0 +1,252 @@
+    response_data = {"userUid": _TEST_USER_UID, "userId": _TEST_USER_LEGACY_ID}
+    mock_py42_client.users.create_user.return_value = create_mock_response(
+        mocker, response_data
+    )
+    return mock_py42_client
+
+
+@fixture
+def mock_py42_with_legacy_id_user(mocker, mock_py42_client):
+    response_data = {
+        "users": [{"userUid": _TEST_USER_UID, "userId": _TEST_USER_LEGACY_ID}]
+    }
+    mock_py42_client.users.get_by_username.return_value = create_mock_response(
+        mocker, response_data
+    )
+    return mock_py42_client
+
+
+@fixture
+def mock_py42_with_user_not_found(mocker, mock_py42_client):
+    mock_py42_client.users.get_by_username.return_value = create_mock_response(
+        mocker, {"totalCount": 0, "users": []}
+    )
+    return mock_py42_client
+
+
+class TestCode42UsersConnector(object):
+    def test_handle_action_when_creating_user_calls_with_expected_args_and_sets_success_status(
+        self, mock_py42_with_create_user
+    ):
+        param = {
+            "org_uid": _TEST_ORG_UID,
+            "username": _TEST_USERNAME,
+            "password": _TEST_PASSWORD,
+            "first_name": _TEST_FIRSTNAME,
+            "last_name": _TEST_LASTNAME,
+            "notes": _TEST_NOTE,
+        }
+        connector = create_fake_connector("create_user", mock_py42_with_create_user)
+        connector.handle_action(param)
+        mock_py42_with_create_user.users.create_user.assert_called_once_with(
+            org_uid=_TEST_ORG_UID,
+            username=_TEST_USERNAME,
+            email=_TEST_USERNAME,
+            password=_TEST_PASSWORD,
+            first_name=_TEST_FIRSTNAME,
+            last_name=_TEST_LASTNAME,
+            notes=_TEST_NOTE,
+        )
+        assert_success(connector)
+
+    def test_handle_action_when_creating_user_and_is_successful_sets_success_message(
+        self, mock_py42_with_create_user
+    ):
+        param = {"org_uid": _TEST_ORG_UID, "username": _TEST_USERNAME}
+        connector = create_fake_connector("create_user", mock_py42_with_create_user)
+        connector.handle_action(param)
+        expected_message = (
+            f"{_TEST_USERNAME} was created with user_id: {_TEST_USER_UID}"
+        )
+        assert_successful_message(connector, expected_message)
+
+    def test_handle_action_when_creating_user_adds_response_to_data(
+        self, mock_py42_with_create_user
+    ):
+        param = {"org_uid": _TEST_ORG_UID, "username": _TEST_USERNAME}
+        connector = create_fake_connector("create_user", mock_py42_with_create_user)
+        connector.handle_action(param)
+        assert_successful_single_data(
+            connector, {"userUid": _TEST_USER_UID, "userId": _TEST_USER_LEGACY_ID},
+        )
+
+    def test_handle_action_when_creating_user_without_password_param_generates_password(
+        self, mocker, mock_py42_with_create_user
+    ):
+        mock_passgen = mocker.patch("code42_connector._generate_password")
```

Comment:
Instead of this, you can test that the generated password contains a capital letter, a lower case letter, is of a certain length, a symbol, and a number. That way, we are at least testing that it meets the criteria without caring about its actual value